scripts/from-src/get-properties.js

```diff
@@ -0,0 +1,74 @@
+module.exports = function(componentName, content) {
+	const srcProperties = getSrcProperties(content);
+	const defaults = getDefaultValues(content);
+	const properties = eval(`(${srcProperties})`);
+	return getPropertiesObject(properties, defaults, srcProperties);
+};
+
+function getSrcProperties(fileContent) {
+	let properties = fileContent.split('properties() {return ');
+	if (properties.length === 1) {
+		return "{ '': '' }"; // fix tablemark error for empty object
+	}
+	properties = properties[1].split(';')[0];
+	return properties;
+}
+
+function getPropertiesObject(properties, defaultValues, srcProperties) {
+	const outObject = [];
+	Object.keys(properties).forEach((propertyName, index) => {
+		const propertyData = properties[propertyName];
+		delete propertyData.reflect; // delete reflect since it's not useful information for readme
+		outObject[index] = {
+			Property: propertyName,
+			Attribute: propertyData.attribute || propertyName,
+			Type: getPropertyType(propertyData),
+			Default: getDefaultValueOfProperty(propertyName, propertyData, defaultValues),
+			Description: getPropertyDescription(propertyName, srcProperties)
+		};
+	});
+	return outObject;
+}
+
+function getDefaultValues(fileContent) {
+	const defaultsContent = fileContent.replace(/;/g, '').replace(/ /g, '');
+	let defaults = defaultsContent.split('constructor(){super()')[1];
+	if (defaults === undefined) {
+		return;
+	}
+	defaults = defaults.split('}')[0];
+	defaults = defaults.split('this.').map(data => data.split('='));
+	defaults.shift();
+
+	defaults = defaults.filter(item => item.length === 2);
+	return defaults.reduce((o, k) => ((o[k[0]] = k[1]), o), {});
```

Comment:
I do not completely understand, what is happening in this line of code. Could you add some comments, or use more descriptive names and probably split it to separate functions?

Reply:
I refactored it to be clear.